android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/AllFilter.kt

```diff
@@ -0,0 +1,762 @@
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Divider
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.RangeSlider
+import androidx.compose.material3.SliderDefaults
+import androidx.compose.material3.SwitchDefaults
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberDateRangePickerState
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.components.bottomsheet.DatePickerBottomSheet
+import org.uwaterloo.subletr.components.button.DateInputButton
+import org.uwaterloo.subletr.components.switch.PrimarySwitch
+import org.uwaterloo.subletr.enums.Gender
+import org.uwaterloo.subletr.enums.HousingType
+import org.uwaterloo.subletr.pages.account.PageDivider
+import org.uwaterloo.subletr.pages.home.HomePageUiState
+import org.uwaterloo.subletr.theme.filterBoldFont
+import org.uwaterloo.subletr.theme.filterRegularFont
+import org.uwaterloo.subletr.theme.subletrPalette
+import org.uwaterloo.subletr.utils.displayDateFormatter
+import org.uwaterloo.subletr.utils.parseUTCDateTimeToLocal
+import org.uwaterloo.subletr.utils.storeDateFormatISO
+import java.text.SimpleDateFormat
+import java.time.ZoneOffset
+import java.util.Date
+import java.util.Locale
+
+@SuppressWarnings("CyclomaticComplexMethod","LongMethod","ComplexCondition")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AllFilter(
+	currentFilterVals: HomePageUiState.FiltersModel,
+	updateFilterVals: (HomePageUiState.FiltersModel) -> Unit,
+	coroutineScope: CoroutineScope,
+	closeAction: () -> Unit,
+) {
+	val listState: LazyListState = rememberLazyListState()
+	var housingPref by remember {
+		mutableStateOf(
+			currentFilterVals.housingType
+		)
+	}
+	var startDateButtonText by remember {
+		mutableStateOf(
+			currentFilterVals.dateRange.startingDate?.let {
+				parseUTCDateTimeToLocal(it)
+			} ?: ""
+		)
+
+	}
+	var endDateButtonText by remember {
```

Comment:
Code is copy pasted from other filters, will refactor it if have more time in the end